Enabled Gradle enterprise, added docs

mockitoguy opened the pull request:
Enabled Gradle enterprise plugin. This way, when running with "--scan" flag (like we do on CI) will correctly publish a build scan.

Also added some documentation.

mockitoguy committed:
Enabled Gradle enterprise, added docs

koral-- commented on settings.gradle:
```diff
@@ -1 +1,12 @@
+gradleEnterprise {
+    buildScan {
+        termsOfServiceUrl = "https://gradle.com/terms-of-service"
+        termsOfServiceAgree = "yes"
```
We should add something like this: 
`isUploadInBackground = System.getenv("CI") == null`

https://docs.gradle.com/enterprise/gradle-plugin/#configuring_background_uploading

mockitoguy committed:
Improved upload reliability

More info: https://docs.gradle.com/enterprise/gradle-plugin/#configuring_background_uploading
